### ISSUE TITLE
Updates for casadi 3.6

### ISSUE THREAD

**Issue by tinosulzer** [pybamm-team/PyBaMM#2858]
Release notes: https://github.com/casadi/casadi/releases/tag/3.6.0

Will add as found

- time grid points should be passed when creating interpolator instead of when solving
- rootfinder errors

**Comment by tinosulzer** [comment]
I don't know how to fix the failing sensitivity test so skipping for now, @martinjrobins @jsbrittain could you take a look at this?

**Comment by martinjrobins** [comment]
sure, this is also reported in https://github.com/pybamm-team/PyBaMM/issues/2789#issuecomment-1482739316.

**Comment by tinosulzer** [comment]
I think it might be a different error here

**Comment by martinjrobins** [comment]
Did PR #2883 fix the problem? I just tried `test_sensitivities` on spm, spme and dfn with idaklu solver and it seemed to work ok.

**Comment by martinjrobins** [comment]
ahh ok, I wasn't aware you'd added that

**Comment by martinjrobins** [comment]
near as I can figure this is the problem. Say you have a model like this

$$\frac{du}{dt} = - a * u$$

$$ v = u $$

$$ u(0) = 1$$

$$ v(0) = 1$$

For $a=1$, sundials calculates the initial condition for $s = \frac{dv}{da}$ as $s(0) = [0, 0]$, presumably because v(0) does not depend on $a$, but if you correctly write the initial condition for $v$ as:

$$\frac{du}{dt} = - a * u$$

$$ v = u $$

$$ u(0) = 1$$

$$ v(0) = a$$

Then sundials finds the initial condition should be $s(0) = [0, 1]$. Not exactly sure what the expected behaviour is really, or why sundials isn't calculating the correct ic for $s$ even when you don't give the correct symbolic form for $v(0)$

**Comment by tinosulzer** [comment]
$s(0) = [0, 1]$ is the correct initial condition in the second case, right?

**Comment by martinjrobins** [comment]
> s(0)=[0,1] is the correct initial condition in the second case, right?

yup, and I believe it should be for the 1st as well, so that is why I'm a bit confused that the `calcic` function in sundials (ie. calculate consistent initial conditions) isn't finding it, it just returns `[0, 0]` for the 1st and `[0, 1]` for the 2nd.

**Comment by martinjrobins** [comment]
ahh ok. You have to call `IDAGetSens` after `IDACalcIC` in order for the solution to actually show up in your sensitivity vector..... fixed now
